### DISCUSSION TITLE
Pass through and test work

### DISCUSSION THREAD

**Discussion by Numiah** [bitfieldlabs/afterglow#22] [Q&A] [answered]
A quick debug question on a System11c Bad Cats. (Hat Bender used)
When set to 1, playfield lights work in the random test pattern.
Setting to 4 pass through, all lights work and get addressed as original.
Setting to 3 gives a quick flash from bottom playfieltd to top playfield, but then lights stay out.
All switches off result in no working lights.
What could make it think there's no valid signal? With passtrough working I assume everything is at least connected correctly?
Thank you in advance!

**Comment by bitfieldlabs** [comment]
Hmm. What does the STATUS LED do?

There are two options to debug this further:

1. You connect to the USB port with a serial monitor and check the output
2. You record the input, download it from the AG board and send it to me. Instructions are [here](https://github.com/bitfieldlabs/afterglow/wiki/Replay-Mode#recording-data).

**Comment by Numiah** [comment]
Sorry for the delay.
I am still struggling with the steep learning curve on how to extract the recorded file. (libusb, cmake, mingw, VSC, GIT, PicoTool and others)
Although I'm pretty savvy, still I get dizzy when one install leads to two other installs...

For now, the status led works only when switch is set to 1 or when it is recording.
When it switches to replay after recording the led stays off and no controlled lights are lit.
When in normal use, the status led remains off.

**Reply by bitfieldlabs** [reply]
But you didn't manage to download the recording with picotool?

**Comment by Numiah** [comment]
Not yet as I am struggling installing/using picotool...

D:\PicoTool\build>picotool save -r 0x10040000 0x103c4000 recording.bin
No accessible RP2040 devices in BOOTSEL mode were found.
but:
Device at bus 10, address 1 appears to be a RP2040 device with a USB serial connection, not in BOOTSEL mode. You can
force reboot into BOOTSEL mode via 'picotool reboot -f -u' first.
D:\PicoTool\build>picotool reboot -f -u
ERROR: Unable to access device to reboot it; Make sure there is a driver installed via Zadig
So it appears I now need to get Zadig and find out how that works. DL-ed Zadig, replaced drivers and get another error : "
ERROR: Unable to locate reset interface on the device
Or I get : ERROR: Failed to get descriptor -5
Sorry for the inconvenience sir. I am doing my best ;)

**Reply by bitfieldlabs** [reply]
Sorry, cannot help much with this as I'm doing everything on Linux..

**Reply by Numiah** [reply]
No worries. I'll report back when I have the file.
I'll run a Linux distro this weekend.

**Comment by bitfieldlabs** [comment]
I have added support for recording download to the [Afterglow Config Tool](https://github.com/bitfieldlabs/afterglow/wiki/Configuration-Software).  This should be easier than using picotool.
For this to work you have to update the firmware of the board to the newest version 3.03 as well.

**Comment by Numiah** [comment]
Thank you so much for helping out!
I tried, but the AG is not showing up as a COM port, but as a USB device. Thus the config tool can't connect to it since I can only select COM-ports in the tool. Yikes, I feel so dumb sometimes.

**Reply by bitfieldlabs** [reply]
Hmm. Which Windows version? I have tried on different machines, and the board always shows up as serial COM port when plugged in normally (not in BOOTSEL mode).

**Comment by Numiah** [comment]
Success !! Just when I was wondering why life was giving me lemons..
Tried different Win10 and there it was. Sigh.
I had to zip it to be allowed to upload.
[recording.zip](https://github.com/user-attachments/files/17880072/recording.zip)

**Comment by bitfieldlabs** [comment]
I can see that in your recording the fourth column is always ON. This makes the afterglow reject the data as invalid input.

Check the column 4 line in your setup (R14, R32, D12). Do you have another afterglow board to test with?

**Comment by Numiah** [comment]
Thank you for looking into it.
I swapped boards they all have the same behavior. Ditto for the HAT's.
However..... When measuring voltages on the columns, all are floating at ±5v, except col4, which is at16v. So is the TIP42 driving it.
When connecting a lamp directly between any row and column4 (or any other for that matter) everything behaves normally and blinks in all lamp test mode. Not oddly bright even.
I suspect the TIP42 driving col4 is the culprit. It doesn't measure shorted, but reads different though.
I'll replace it and report back in a few days.

**Comment by bitfieldlabs** [comment]
Hmm. If you had such an issue on the driver board you should also have bad ghosting when connected without the afterglow board. Did you measure with the AG board connected?

**Comment by Numiah** [comment]
I measured with nothing connected, AG connected and with everything connected the original way.
Funny thing is, the abnormal ghosting got me looking into OCD and Afterglow in the first place. There is no ghosting with regular bulbs btw.
Everything works as it should. 
I swapped this AG with my other inside a WPC. Over there everything is as it should. 
Most likely the AG discovered a fault on my driverboard lol. I won't swap transistors to see the difference, I'll just shotgun that transistor as it reads funny. (in circuit)
Besides that I am installing Linux on a laptop and will be doing a deep dive into learning to manage this Afterglow.
I'll keep you posted.

**Comment by Numiah** [comment, marked as answer]
Success! After all those efforts it indeed turned out the transistor that was 50% broken!
I would never ever have found this problem without your help and endless patience.
Thank you so much for sticking with me and spending your precious time on this 'issue'.
*bows*

btw, is there a place where I can donate a few beers to you? PayPal?

**Comment by bitfieldlabs** [comment]
Glad to hear it works! I may try to indicate/handle such things in "smart mode" in the future.

Ehm, haven't thought about donations yet. Let me think...  ;-)
